### ISSUE TITLE
No popular distro live image boots

### ISSUE THREAD

**Issue by qu1ck** [netbootxyz/netboot.xyz#1384]
**Describe the bug**
Live images seem to be completely broken for a good chunk of popular distros.
I tried:
* Debian 12 kde
* Debian 12 cinnamon
* Fedora 39 kde
* Ubuntu 22.04 xfce
* Ubuntu 23.04 kde

Not one of them boots, most fail with unable to mount one thing or another.

**To Reproduce**
Boot any of the above distros

**Expected behavior**
I should be able to boot something modern

**Comment by OptimusGREEN** [comment]
I can't get any of the live cd's to work.

**Comment by jkoelker** [comment]
The listed distros boot currently with the exception of KDE fedora (which appears to be a dracut issue not requesting the squashfs image correctly, but that's been the case for at least the last two releases), the issue is you probably have not allocated enough ram for the distro's requirements for the live image. Unfortunately over the years distros livecd images have steadily grown.

I have just validated each with 8G:

```
qemu-system-x86_64 -bios /usr/share/edk2-ovmf/x64/OVMF.fd -cdrom netboot.xyz.iso -m 8G -smp $(nproc)
```

**Comment by OptimusGREEN** [comment]
Thanks for the reply.

I can confirm that raising to 8gb ram indeed  fixed my issue. great!

**Comment by antonym** [comment]
Thanks @jkoelker, the Live Images do usually just need a lot more memory unfortunately.

There also may be some issues on Fedora that I need to circle back on, because they have steadily grown, I'll need to add the rootfs split logic into the initrds that split out as well:

https://github.com/netbootxyz/netboot.xyz/issues/1214

**Comment by qu1ck** [comment]
This is what I get on 10GB VM, same result on 16 GB

Ubuntu 23.04 KDE
![image](https://github.com/netbootxyz/netboot.xyz/assets/3622532/2a884b35-1d63-4b27-b3d0-32497005c12e)

Ubuntu 22.04 xfce
![image](https://github.com/netbootxyz/netboot.xyz/assets/3622532/bc297d58-a7de-4cfc-b256-7458150393fd)

Debian 12 cinnamon
![image](https://github.com/netbootxyz/netboot.xyz/assets/3622532/8080c4c9-216f-45b3-8a19-d171bd233c09)

Debian 12 KDE
![image](https://github.com/netbootxyz/netboot.xyz/assets/3622532/5cf21091-b39c-4853-b079-0478a92a6e05)

What am I doing wrong?

**Comment by antonym** [comment]
Make sure you are on the newest rev too, either 2.0.75 or rolling boot.netboot.xyz, 2.0.74 had some some bad images that may have been missing curl.  I just tried Ubuntu 23.04 KDE and it loaded fine.

<img width="877" alt="Screenshot 2023-12-27 at 11 52 28 PM" src="https://github.com/netbootxyz/netboot.xyz/assets/245007/1f7bd668-cfc2-49f5-a24d-2c73433acc26">

**Comment by qu1ck** [comment]
Yes, I'm on 2.0.75. Netinst images load fine and same VM loads from live iso's fine.

**Comment by greatwolf** [comment]
I'm experiencing the same issue as @qu1ck when booting Linux Mint LMDE edition. This is on a laptop with 8gb of ram. His Deb 12 Cinnamon bootup screenshot is what I'm seeing on my laptop during live boot attempt.

**Comment by jkoelker** [comment]
When ya'll get dumped to the shell, what does `/proc/cmdline` say? Are you using the github version of everything or are you trying to do self-hosting? If the latter try using the hosted version first, if that works, then its something in your self-hosted setup.

**Comment by qu1ck** [comment]
I am using the official netboot.xyz docker image and I'm caching the init image/squashfs files locally to speed up my boot times.

And you are right, sticking a netboot.xyz.iso into my vm and booting from it I was able to boot into debian 12 cinnamon live image.

Trying some things I was able to boot from my self hosted netboot.xyz if I dropped the cache and reverted config changes that pointed to cache host instead of github downloads.

After some trial and error I found that I didn't cache the "default" squashfs which is not obvious that it is needed. After caching it as well I can boot debian 12 and ubuntu 23.04 but not ubuntu 23.10.

Still, better than nothing.
